dask/sizeof.py

```diff
@@ -264,12 +265,17 @@ def register_xarray():
 
     import xarray as xr
 
+    XARRAY_VERSION = Version(xr.__version__)
+    XARRAY_GE_2024_02 = XARRAY_VERSION >= Version("2024.02.0")
+
     @sizeof.register(xr.DataArray)
-    @sizeof.register(xr.NamedArray)
```

Comment:
Sorry for the delay here. Using `Variable` here might be enough? (Variable is a subclass of the new NamedArray, and if you need NamedArray, then you need Variable on older versions).